stumpy/core.py

```diff
@@ -1576,6 +1604,16 @@ def mass(
     if np.any(~np.isfinite(Q)):
         distance_profile[:] = np.inf
     else:
+        if query_idx is not None:
+            query_idx = int(query_idx)
+            if not np.allclose(Q, T[query_idx : query_idx + m]):
+                msg = (
```

Comment:
Note 1) This warning cannot be outside of the if/else block (i.e. above  the line`if np.any(~np.isfinite(Q))`). This is because in such case, the following inputs will result in warning:
```
Q = [np.nan, np.nan, np.nan]
T = [np.nan, np.nan, np.nan, 0, 1, 2]
query_idx = 0
```

Note 2) This warning message must be before calling `core.preprocess(...)` since we need to have the original `T` here. If we put this warning at the end (i.e. after preprocesssing `T`), then the following input does not result in warning.

```
Q = [0, 0, 0, 1]
T = [np.nan, np.nan, np.nan, 1]
query_idx = 0
```

Reply:
As a follow up to my previous comment, what if the user provide preprocess `T` and `M_T` in the input. Then, the last example in previous example still results in wrong warning.

Reply:
Technically, the comparison can be done very early on:

```
check_window_size(m, max_size=Q.shape[-1])
if query_idx is not None:
    query_idx = int(query_idx)
    QT_query_idx_allclose = np.allclose(Q, T[query_idx : query_idx + m])
T = _preprocess(T)
```

Then, later:

```
    if np.any(~np.isfinite(Q)):
        distance_profile[:] = np.inf
    else:
        if query_idx is not None and not QT_query_idx_allclose:  # pragma: no cover
                msg = (
```

Reply:
nice! Thank you for the suggestion!!